app/configurator/configurator-state.tsx

```diff
@@ -1410,6 +1416,68 @@ const reducer: Reducer<ConfiguratorState, ConfiguratorStateAction> = (
 
       return draft;
 
+    case "DATASET_ADD":
+      if (draft.state === "CONFIGURING_CHART") {
+        for (const chartConfig of draft.chartConfigs) {
+          chartConfig.cubes[0].joinBy = action.value.joinBy.left;
+          chartConfig.cubes.push({
+            iri: action.value.iri,
+            joinBy: action.value.joinBy.right,
+            filters: {},
+          });
+
+          // Need to go over fields, and replace any IRI part of the joinBy by "joinBy"
+          const fields = Object.values(chartConfig.fields);
+          while (fields.length > 0) {
+            const f = fields.pop();
+            for (const fieldName of [
+              "componentIri",
+              "leftComponentIri",
+              "rightComponentIri",
+            ]) {
```

Comment:
Can we have these fields? Were they added in some prior commit? I searched in codebase and I couldn't find these strings 👀

Reply:
Thanks, again good catch, the fields had a typo. I consolidated the fields and extracted them in config-types, where they are defined. I still find it brittle though.